### PR TITLE
feat: add model for LearnerPathwayStep | Work in progress

### PR DESCRIPTION
**Pull request by muneebGH**

### CODE REVIEW

File: course_discovery/apps/learner_pathway/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Generated by Django 3.2.8 on 2021-11-25 14:44
```
Comment (saleem-latif) [inline course_discovery/apps/learner_pathway/migrations/0001_initial.py:1]: I think you need to run `makemigrations` again, the changes in `LearnerPathwayNode` are not showing up in migrations.

Please delete this migration file a run `makemigrations` again.

**Review by saleem-latif — APPROVED** [review]
One comment, once that is addresses 👍🏼

File: course_discovery/apps/learner_pathway/migrations/0001_initial.py (hunk 1 above)
Comment (saleem-latif) [reply course_discovery/apps/learner_pathway/migrations/0001_initial.py:1]: Also make sure to squash commits before merging.

File: course_discovery/apps/learner_pathway/models.py [hunk 2]
```diff
@@ -32,3 +33,45 @@ def get_skills(self) -> [str]:
...
+        if node:
+            node.delete()
+
+    def get_estimated_time_of_completion(self):
+        return sum([node.get_estimated_time_of_completion() for node in self.get_nodes()])
+
+    def get_skills(self):
+        skills = set()
+        for node in self.get_nodes():
+            skills.update(node.get_skills())
```
Comment (muhammad-ammar) [inline course_discovery/apps/learner_pathway/models.py]: Please note, course and program subclasses will return a list of dicts where each dict will contain skill name and skill description.

**Review by muhammad-ammar — APPROVED** [review]
LGTM. Please see my comment.

File: course_discovery/apps/learner_pathway/migrations/0001_initial.py (hunk 1 above)
Comment (saleem-latif) [reply course_discovery/apps/learner_pathway/migrations/0001_initial.py:1]: Sorry, forgot that `LearnerPathwayNode` is a meta model.

File: course_discovery/apps/learner_pathway/models.py (hunk 2 above)
Comment (muneebGH) [reply course_discovery/apps/learner_pathway/models.py]: done

**Pull request merged**